src/ProductConstructionService/ProductConstructionService.ReproTool/ReproTool.cs

```diff
@@ -0,0 +1,280 @@
+                await ghClient.Repository.Commit.Get(sourceRepoOwner, sourceRepoName, options.Commit);
+            }
+            catch (NotFoundException)
+            {
+                throw new ArgumentException($"Commit {options.Commit} doesn't exist in repo {subscription.SourceRepository}");
+            }
+            sourceRepoSha = options.Commit;
+        }
+
+        // Create the temp channel
+        var channelName = Guid.NewGuid().ToString();
+        await using var channel = await darcProcessManager.CreateTestChannelAsync(channelName);
+
+        // The build
+        var build = await CreateBuildAsync(
+            isForwardFlow ? productRepoForkUri : VmrForkUri,
+            isForwardFlow ? defaultChannel.Branch : vmrTmpBranch.Value,
+            sourceRepoSha);
+
+        // And the subscription
+        await using var testSubscription = await darcProcessManager.CreateSubscriptionAsync(
+            channel: channelName,
+            sourceRepo: isForwardFlow ? productRepoForkUri : VmrForkUri,
+            targetRepo: isForwardFlow ? VmrForkUri : productRepoForkUri,
+            targetBranch: isForwardFlow ? vmrTmpBranch.Value : subscription.TargetBranch,
+            sourceDirectory: subscription.SourceDirectory,
+            targetDirectory: subscription.TargetDirectory);
+
+        // Add it to the channel
+        await darcProcessManager.AddBuildToChannelAsync(build.Id, channelName);
+
+        // Trigger the subscription
+        await TriggerSubscriptionAsync(testSubscription.Value);
+
+        logger.LogInformation("Code flow successfully recreated. Press enter to finish and cleanup");
+        Console.ReadLine();
+
+        // Cleanup
+        await DeleteAllDarcPRBranchesAsync(isForwardFlow ? VmrForkRepoName : productRepoUri.Split('/').Last());
+    }
+
+    private async Task DeleteAllDarcPRBranchesAsync(string repo)
+    {
+        var branches = await ghClient.Repository.Branch.GetAll(MaestroAuthTestOrgName, repo);
+        foreach (var branch in branches.Where(b => b.Name.StartsWith(DarcPRBranchPrefix)))
+        {
+            await DeleteGitHubBranchAsync(repo, branch.Name);
+        }
+    }
+
+    private async Task AddRepositoryToBarIfMissingAsync(string repositoryName)
+    {
+        if ((await context.Repositories.FirstOrDefaultAsync(repo => repo.RepositoryName == repositoryName)) == null)
+        {
+            context.Repositories.Add(new Maestro.Data.Models.Repository
+            {
+                RepositoryName = repositoryName,
+                InstallationId = InstallationId
+            });
+            await context.SaveChangesAsync();
+        }
+    }
+
+    private async Task<Build> CreateBuildAsync(string repositoryUrl, string branch, string commit)
+    {
+        Build build = await pcsApi.Builds.CreateAsync(new BuildData(
+            commit: commit,
+            azureDevOpsAccount: "test",
+            azureDevOpsProject: "test",
+            azureDevOpsBuildNumber: "1",
```

Comment:
Can this be something that follows the usual format a bit more?

Maybe something like `YYYYMMDD.R` where `R` is a random number between 1 and 75?

Reply:
added, altho I think this is only seen in the PR if even there, so it's not very important